waddrmgr/address.go

```diff
@@ -98,6 +101,7 @@ type managedAddress struct {
 	internal         bool
 	compressed       bool
 	used             bool
+	watchingOnly     bool
 	pubKey           *btcec.PublicKey
 	privKeyEncrypted []byte
```

Comment:
Why do we need a watchingOnly boolean here when the privKeyEncrypted field could be nilled instead?